content/docs/user-guide/project-structure/pipelines-files.md

```diff
@@ -0,0 +1,420 @@
+Python parameters file (`params.yaml` by default). Example:
+
+```yaml
+stages:
+  preprocess:
+    cmd: ./clean.sh raw.txt clean.txt
+    deps:
+      - raw.txt
+    params:
+      - threshold
+      - passes
+    outs:
+      - clean.txt
+```
+
+This allows several stages to depend on values of a shared structured file
+(which can be versioned directly with Git). See also `dvc params diff`.
+
+### Metrics and Plots outputs
+
+Like [common outputs](#outputs), <abbr>metrics</abbr> and <abbr>plots</abbr>
+files are produced by the stage `cmd`. However, their purpose is different.
+Typically they contain metadata to evaluate pipeline processes. Example:
+
+```yaml
+stages:
+  build:
+    cmd: python train.py
+    deps:
+      - features.csv
+    outs:
+      - model.pt
+    metrics:
+      - accuracy.txt:
+          cache: false
+    plots:
+      - auc.json:
+          cache: false
+```
+
+> `cache: false` is typical here, since they're small enough for Git to version
+> directly.
+
+The commands in `dvc metrics` and `dvc plots` help you display and compare
+metrics and plots.
+
+## Templating
+
+⚠️ This feature is only available in DVC 2.0 ⚠️
+
+`dvc.yaml` supports a templating format to insert values from different sources
+in the YAML structure itself. These sources can be
+[parameters files](/doc/command-reference/params), or `vars` defined in
+`dvc.yaml` instead.
+
+> Note that this parameterization feature is only supported via manual edition
+> of `dvc.yaml` and incompatible with `dvc run`.
+
+Let's say we have `params.yaml` (default params file) with the following
+contents:
+
+```yaml
+models:
+  us:
+    threshold: 10
+    filename: 'model-us.hdf5'
+```
+
+Those values can be used anywhere in `dvc.yaml` with the `${}` _substitution
+expression_:
+
+<!-- prettier-ignore-start -->
+```yaml
+stages:
+  build-us:
+    cmd: >-
+      python train.py
+      --thresh ${models.us.threshold}
+      --out ${models.us.filename}
+    outs:
+      - ${models.us.filename}:
+        cache: true
+```
+<!-- prettier-ignore-end -->
+
+DVC will track simple param values (numbers, strings, etc.) used in `${}` (they
+will be listed by `dvc params diff`).
+
+Alternatively, values for substitution can be listed as top-level `vars` like
+this:
+
+```yaml
+vars:
+  - models:
+      us:
+        threshold: 10
+  - desc: 'Reusable description'
+
+stages:
+  build-us:
+    desc: ${desc}
+    cmd: python train.py --thresh ${models.us.threshold}
+```
+
+> DVC merges values from params files and `vars` when possible. For example,
+> `{"grp": {"a": 1}}` merges with `{"grp": {"b": 2}}`, but not with
+> `{"grp": {"a": 7}}`.
+
+> Note that values from `vars` are not tracked like parameters.
+
+To load additional params files, list them in the top `vars`, in the desired
+order, e.g.:
+
+> Their paths will be evaluated based on
+> [`wdir`](/doc/user-guide/dvc-files/dvc.yaml#accepted-fields), if one given.
```

Comment:
This note should be moved on "Stage-specific values" below.

Reply:
> [note on wdir] should be moved on "Stage-specific values

You mean also mentioned in that part? Because it's important here too (and this part comes first). For now I made it more general so people can assume it applies to local vars too.